docs/dev/howtoTestRepository.md

```diff
@@ -0,0 +1,156 @@
+```
+  @Test
+  public void getInstanceXml() throws Exception {
+  // Getting Xml from a API, checking if its equal to the expected local file.
+  }
+  @Test
+  public void getServicetemplate() throws Exception {
+  // Getting JSON from a API, checking if its equal to the expected local file.
+  }
+```
+
+These are the two basic methods to test the capability of the API, to serve JSON and XML-Filetypes as response.
+
+## Things to prepare on Test
+Before running the Basic Test logic like in the Chapter [Add Test Methods](##Add Test Methods)
+you have to get a Github commit-id from the [Test-Repository](https://github.com/winery/test-repository).
+
+In the following example is shown how to get a revision:
+```
+  @Test
+  public void getServicetemplate() throws Exception {
+    this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+    // Test Logic 
+  }
+```
+## Implement a Testing Logic
+
+To Test the Logic you can simply implement this two liner Test, it will check out the Revision `c25aa724201824fce6eddcc7c35a666c6e015880` from Github Repository located under [Test-Repository](https://github.com/winery/test-repository).
+After wards it will send a `GET`Request against the Backend initialized by the GIT-Repository Data. The Response will be tested for equality against a File with the expected Result.
+  
+If the Result is equal the Test will become marked as successful.  
+```
+	@Test
+ 	public void getInstanceXml() throws Exception {
+ 		this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+ 		this.assertGet(INSTANCE_URL, INSTANCE_LOCAL_PATH);
+ 	}
+```
+## Run a Test or Test Class
+If you want to run a Test you can simply Run it from Menu in your Eclipse or IntelliJ IDE.
+
+## Appendix 
+
+Here we have a simple implementation of the `NodeTypeImplementationResourceTest`. 
+````
+
+package org.eclipse.winery.repository.resources.entitytypeimplementations.nodetypeimplementations;
+
+import org.eclipse.winery.repository.resources.AbstractResourceTest;
+
+import org.junit.Test;
+
+public class NodeTypeImplementationResourceTest extends AbstractResourceTest {
+	private final String ENTITY_TYPE = "nodetypeimplementations/";
+	private final String INSTANCE_XML_PATH = "entityimplementations/" + ENTITY_TYPE + "instance.xml";
+	private final String BAOBAB_JSON_PATH = "entityimplementations/" + ENTITY_TYPE + "baobab_inital.json";
+
+	public static final String FOLDERPATH = "http%3A%2F%2Fwinery.opentosca.org%2Ftest%2Fnodetypeimplementations%2Ffruits/baobab_impl";
+
+
+	private final String INSTANCE_URL = ENTITY_TYPE + FOLDERPATH;
+
+	@Test
+	public void getInstanceXml() throws Exception {
+		this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+		this.assertGet(testStringConverter(INSTANCE_URL), INSTANCE_XML_PATH);
+	}
+
+	@Test
+	public void getServicetemplate() throws Exception {
+		this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+		this.assertGet(ENTITY_TYPE, BAOBAB_JSON_PATH);
+	}
+````
```

Comment:
Please add a license as presented at the bottom at https://github.com/eclipse/winery/blob/master/docs/dev/CodeHeaders.md#license

Reply:
done

Reply:
No, not done. Otherwise, this box would disappear.

Please create one commit as described at http://eclipse.github.io/winery/dev/ToolChain#github---preparation-first-pull-request-to-offical-eclipse-winery-repository

Currently, it are multiple commits, which is not OK.

Reply:
done